pkgs/servers/nosql/arangodb/default.nix

```diff
@@ -1,68 +1,60 @@
-{ stdenv, lib, fetchFromGitHub, openssl, zlib, cmake, python2, perl, snappy, lzo, which, catch2, catch }:
-
-let
-  common = { version, sha256 }: stdenv.mkDerivation {
-    pname = "arangodb";
-    inherit version;
+{ gcc10Stdenv
+, git
+, lib
+, fetchFromGitHub
+, openssl
+, zlib
+, cmake
+, python3
+, perl
+, snappy
+, lzo
+, which
+}:
+
+gcc10Stdenv.mkDerivation rec {
```

Comment:
There's an explicit warning in the cmake:

```
CMake Warning at CMakeLists.txt:616 (message):
  ArangoDB requires g++ 11.0 or newer, building with older compiler versions
  is unsupported
```

Reply:
I see that but it does seem to compile fine. I'm not sure if they mean that gcc11 is required or that anything else is untested.

Reply:
But why fix it to gcc10 in the first place? You haven't explained this.

Reply:
You're right! Thank you! Added some commentary.

Reply:
Do you think gcc10 is acceptable? I'm not sure what to make of the message (particularly the bit about being "unsupported")

Reply:
gcc10 seems to work, but the point is, why override it from gcc11, which is the default in nixpkgs?

Reply:
Oh! It's just my error! I thought I had checked the stdenv.cc version and apparently I didn't read closely enough.  I'll change this today :)